### PR TITLE
feat: [FFM-6649]: Basic caching

### PR DESCRIPTION
**Pull request by knagurski**
This PR adds a new basic caching ability, using `localStorage` as a storage mechanism.

### CODE REVIEW

File: src/cache.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  return 'HARNESS_FF_CACHE_' + targetIdentifier
+}
+
+export function loadFromCache(targetIdentifier: string): Evaluation[] {
+  const cachedEvaluations = window.localStorage.getItem(getCacheId(targetIdentifier))
+
+  if (cachedEvaluations) {
+    try {
+      return JSON.parse(cachedEvaluations)
+    } catch (e) {}
```
Comment (crystal-fong) [inline src/cache.ts:13]: what should it do with any error it catches?

Comment (knagurski) [reply src/cache.ts:13]: It should shut up and keep going

Comment (knagurski) [reply src/cache.ts:13]: This will only happen if the value is not valid JSON, so we can assume there are no evals to restore

Comment (crystal-fong) [reply src/cache.ts:13]: ^ what I tell myself everyday

Comment (knagurski) [reply src/cache.ts:13]: Words to live by!

File: src/cache.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export function removeCachedEvaluation(targetIdentifier: string, flagIdentifier: string): void {
+  const cachedEvals = loadFromCache(targetIdentifier)
+  const existingEvalIndex = cachedEvals.findIndex(({ flag }) => flag === flagIdentifier)
+
+  if (existingEvalIndex > -1) {
+    cachedEvals.splice(existingEvalIndex, 1)
+
+    saveToCache(targetIdentifier, cachedEvals)
+  }
+}
```
Comment (laura-mannus) [inline src/cache.ts:45]: really nice tests

**Review by laura-mannus — APPROVED** [review]

**Pull request merged**